packages/firebase_auth/firebase_auth/ios/Classes/FLTFirebaseAuthPlugin.m

```diff
@@ -481,6 +481,32 @@ - (FIRAuthCredential *)getCredential:(NSDictionary *)arguments {
     credential = [[FIRPhoneAuthProvider providerWithAuth:[self getAuth:arguments]]
         credentialWithVerificationID:verificationId
                     verificationCode:smsCode];
+  } else if ([@"apple.com" isEqualToString:provider]) {
```

Comment:
It seems like we should just allow any providerID here rather than checking for apple.com.

However we might want to consider the case where the developer uses OAuthProvider and passes a providerID that matches one of the cases above. Do you think we've handled it adequately?

What do you think?

```suggestion
  } else {
```

Reply:
I have another versión in which I evaluate that providerId isn't null, idtoken isn't null and ones of accestoken or rawnonce isn't null. 

I think this is better, beacuse the original else can mantein in the code.

I thought that was a mistake, because firebase only support a finite numbers of providers. That's the reason why I descarted that option.

I'm going to send a commit with that, and you can tell my your feed back.